terraform.yml

```diff
@@ -0,0 +1,140 @@
+parameters:
+  - name: original_node_count
+    displayName: "Original Node Count"
+    type: number
+    default: 3
+    values:
+      - 3
+      - 14
+  - name: scale_up_node_count
+    displayName: "Scale Up Node Count +"
+    type: number
+    default: 2
+    values:
+      - 2
+  - name: region
+    displayName: "Region to run pipeline"
+    type: string
+    default: "eastus"
+    values:
+      - "eastus"
+      - "eastus2"
+      - "westus"
+      - "westus2"
+      - "westeurope"
+  - name: run_terraform
+    displayName: "Select terraform example to run"
+    type: string
+    default: "azureblobfiler"
+    values:
+      - "1-filer"
+      - "proxy"
+      - "azureblobfiler"
+  - name: image
+    displayName: "Select vfxt image to run. Defaults to latest."
+    type: string
+    default: "microsoft-avere:vfxt:avere-vfxt-node:6.0.0"
+    values:
+      - "microsoft-avere:vfxt:avere-vfxt-node:6.0.0"
+      - "microsoft-avere:vfxt:avere-vfxt-node:5.3.61"
+      - "microsoft-avere:vfxt:avere-vfxt-node:5.3.51"
+      - "microsoft-avere:vfxt:avere-vfxt-node:5.3.43"
+      - "microsoft-avere:vfxt:avere-vfxt-node:5.3.42"
+  - name: run_vdbench
+    displayName: "Run vdbench - only runs with azureblobfiler"
+    type: boolean
+    default: true
+  - name: vdbench_config
+    displayName: "vdbench config - only runs with azureblobfiler"
+    type: string
+    default: "inmem"
+    values:
+      - "ondisk"
+      - "inmem"
+  - name: run_scale_up
+    displayName: "Run scale-up"
+    type: boolean
+    default: true
+  - name: run_scale_down
+    displayName: "Run scale-down (only runs if scale-up is selected)"
+    type: boolean
+    default: false
+  - name: run_destory_and_cleanup
+    displayName: "Run terraform destroy and rg cleanup."
+    type: boolean
+    default: true
+  - name: pass_in_ssh_key
```

Comment:
You should be able to pass in the SSH key in a text field via parameters. Then you won't need the PUBLIC_KEY variable. I recall you mentioned something wasn't working with parameter substitution for this value.

Maybe having an intermediate variable called PUBLIC_KEY set to the parameter's value would work.

Something like this should work:
```
# Assumes the parameter is called p_ssh_pub_key
variables:
  SSH_PUBLIC_KEY: '${{ parameters.p_ssh_pub_key}}'

jobs:
- job: Terraform_vFXT
```

Also, it'd be good to clarify what is using the public key. Controller, cluster, something else? (I haven't reviewed the rest yet or I'm sure it would be clearer. :-)

Reply:
Had do I get a text field via parameters?

Reply:
found it, using step